.github/workflows/devnet_main_deploy.yml

```diff
@@ -0,0 +1,27 @@
+name: Main CD for Devnet Deployment
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - main
+  push:
+    paths:
+      - "aws/devnet/**"
```

Comment:
So it will redeploy regardless of things changing if merged to `main`, which is probably fine, but also why?

But it will also redeploy on push to any branch whatsoever that changes `aws/devnet/**`, which I don't think is what we want.

Similar comment applies to all other workflows in this PR.